src/main/java/ch/wisv/areafiftylan/users/controller/CurrentUserRestController.java

```diff
@@ -107,15 +106,14 @@
     @JsonView(View.Team.class)
     @RequestMapping(value = "/teams", method = RequestMethod.GET)
     public Collection<Team> getCurrentTeams(Authentication auth) {
-        UserDetails currentUser = (UserDetails) auth.getPrincipal();
-        return teamService.getTeamsByUsername(currentUser.getUsername());
+        User currentUser = (User) auth.getPrincipal();
```

Comment:
I think you can plug in the User directly as a method argument. @MatthijsKok figured this out somewhere. Maybe we can even make it a Class field as every method in this Controller needs the User object.